src/main/java/org/openstreetmap/atlas/geography/atlas/change/FeatureChange.java

```diff
@@ -379,88 +522,145 @@ else if (other.bounds() != null)
+                throw new CoreException("TODO CompletePoint result bounds null, investigate");
             }
             return FeatureChange.add(result);
         }
     }
 
+    private FeatureChange mergeLocationItems_NewlyProposed(final FeatureChange other, // NOSONAR
+            final Map<String, String> mergedTags, final Set<Long> mergedParentRelations)
+    {
+        final AtlasEntity beforeEntityLeft = this.getBeforeView();
+        final AtlasEntity afterEntityLeft = this.getUpdatedView();
+        final AtlasEntity beforeEntityRight = other.getBeforeView();
+        final AtlasEntity afterEntityRight = other.getUpdatedView();
+
+        final MergedMemberBean<Location> mergedLocationBean = FeatureChangeMergers.mergeMember(
+                "location", beforeEntityLeft, afterEntityLeft, beforeEntityRight, afterEntityRight,
+                atlasEntity -> ((LocationItem) atlasEntity).getLocation(), null, null);
+
+        if (afterEntityLeft instanceof Node)
+        {
+            final MergedMemberBean<SortedSet<Long>> mergedInEdgeIdentifiersBean = FeatureChangeMergers
+                    .mergeMember("inEdgeIdentifiers", beforeEntityLeft, afterEntityLeft,
+                            beforeEntityRight, afterEntityRight,
+                            atlasEntity -> ((Node) atlasEntity).inEdges() == null ? null
+                                    : ((Node) atlasEntity).inEdges().stream()
+                                            .map(Edge::getIdentifier)
+                                            .collect(Collectors.toCollection(TreeSet::new)),
+                            FeatureChangeMergers.directReferenceMergerSorted,
+                            FeatureChangeMergers.diffBasedIntegerSortedSetMerger);
+            final MergedMemberBean<SortedSet<Long>> mergedOutEdgeIdentifiersBean = FeatureChangeMergers
+                    .mergeMember("outEdgeIdentifiers", beforeEntityLeft, afterEntityLeft,
+                            beforeEntityRight, afterEntityRight,
+                            atlasEntity -> ((Node) atlasEntity).outEdges() == null ? null
+                                    : ((Node) atlasEntity).outEdges().stream()
+                                            .map(Edge::getIdentifier)
+                                            .collect(Collectors.toCollection(TreeSet::new)),
+                            FeatureChangeMergers.directReferenceMergerSorted,
+                            FeatureChangeMergers.diffBasedIntegerSortedSetMerger);
+            final CompleteNode result = new CompleteNode(getIdentifier(),
+                    mergedLocationBean.getMergedAfterMember(), mergedTags,
+                    mergedInEdgeIdentifiersBean.getMergedAfterMember(),
+                    mergedOutEdgeIdentifiersBean.getMergedAfterMember(), mergedParentRelations);
+            if (result.bounds() == null)
+            {
+                throw new CoreException("TODO CompleteNode result bounds null, investigate");
```

Comment:
What to do here? Before, we had:

```
if (bounds() != null)
{		
    result = result.withAggregateBoundsExtendedUsing(bounds());		
}		
else if (other.bounds() != null)		
{		
    result = result.withAggregateBoundsExtendedUsing(other.bounds());		
}
```

Now, it seems we do not need to handle this.

Reply:
Right now I am throwing an exception to see if this case ever arises as I write more test cases.

Reply:
Ok. I have modified that constructor now, so it is no longer possible for a `CompleteEntity` to ever be unbounded.